src/lib/y2sap/auto.rb

```diff
@@ -175,12 +175,25 @@ def prepare_hana(prod)
         @error = true
         return :abort
       end
-      !prod.key?("sapMDC") || prod["sapMDC"] = "no"
       File.write(@inst_dir + "/ay_q_masterPwd", prod["sapMasterPW"])
       File.write(@inst_dir + "/ay_q_sid",       prod["sid"])
       File.write(@inst_dir + "/ay_q_sapinstnr", prod["sapInstNr"])
-      File.write(@inst_dir + "/ay_q_sapmdc",    prod["sapMDC"])
-      File.write(@inst_dir + "/ay_q_virt_hostname", prod["sapVirtHostname"]) if prod.key?("sapVirtHostname")
+      if prod.key?("sapVirtHostname")
+        File.write(@inst_dir + "/ay_q_virt_hostname", prod["sapVirtHostname"])
+      end
+      xs_routing_mode = prod.key?("xsRoutingMode") ? prod.key?("xsRoutingMode") : "ports"
+      if !["hostname", "ports"].include?(xs_routing_mode)
+        Yast::Popup.Error("Bad XS routing mode: #{xs_routing_mode}. Provided values are 'hostname' or 'ports'")
+        @error = true
+        return :abort
+      end
+      if xs_routing_mode == "hostname" && !prod.key?("xsDomainName")
+        Yast::Popup.Error("If XS routing mode is set to hostname you have to define xsDomainName also")
```

Comment:
also here it looks like old SP6 code without new changes from https://github.com/yast/sap-installation-wizard/pull/38

Reply:
Sorry. I've found the problem.... 
Unfortunately I've merged SLE-15-SP5 instead of SLE-15-SP6 :-(

Reply:
Now I'v merged SP6 into master.
Thank you for your attention!